Update the components colors based on the latest Valet Design System

djalmaaraujo opened the pull request:
## Description

Ps: *Dark mode is not part of the scope*

<img width="2728" alt="image" src="https://user-images.githubusercontent.com/3402/208719572-c68b865c-5ced-4193-a7b1-70f8fb40154f.png">

This PR uses the latest develop/tokens/* files and adapts the components using the new color system. All possible components are being updated.

Preview URL: https://deploy-preview-138--vip-design-system-components.netlify.app/

All changes were made in sync with @donaghokeeffe (VIP Designer), matching the latest Figma files from the Valet Design System.

# Steps to test

1. Go to https://deploy-preview-138--vip-design-system-components.netlify.app/
2. Navigate into all non-deprecated components pages
3. Colors have changed. You should not find any accessibility contrast issues 

<img width="939" alt="image" src="https://user-images.githubusercontent.com/3402/209171090-5b4cd1b0-2552-4ce0-aecb-4246adb12483.png">

# Steps to test (Optional)

1. Go to the vip-dashboard repo
2. Change the `"@automattic/vip-design-system": "0.27.10",` to `"@automattic/vip-design-system": "github:automattic/vip-design-system#f870605e8312c3b90bd4d5981fc8a1f0c85cc98d"`
3. Do `npm install`
4. Do `npm run dev`
5. Test the dashboard works as expected with the new colors

djalmaaraujo committed:
Test updating valet-theme with the latest changes

djalmaaraujo committed:
Refactoring light theme to use valet-theme color system

djalmaaraujo committed:
Improve Notice with the correct color palette

djalmaaraujo committed:
Alias brand to gold

djalmaaraujo committed:
Adjust tabs example

djalmaaraujo committed:
Adjust badge color system

djalmaaraujo committed:
Fix table contrast

djalmaaraujo committed:
Fix Wizard color contrast

djalmaaraujo committed:
Updated tokens from Figma

djalmaaraujo committed:
Notice: adjust colors based on the support theme

djalmaaraujo committed:
Inputs: Adjust colors for default input and label

djalmaaraujo committed:
FormSelect: Adjust colors for the the FormSelect

djalmaaraujo committed:
Fix base input border styles

djalmaaraujo committed:
Autocomplete + Radio: Adjust colors

djalmaaraujo committed:
Checkbox: Adjust styling

djalmaaraujo committed:
Badge: Adjust styling

djalmaaraujo committed:
Normalize colors for components instead of using js vars

djalmaaraujo committed:
Translate input and badge colors to css variables

djalmaaraujo committed:
Remove unused vars

djalmaaraujo committed:
Fix tests for common input styles

djalmaaraujo committed:
Adjusts on links and tabs

djalmaaraujo committed:
Adjust general colors for Toggle, Checkbox, Spinner and deprecated Time and Timeline

djalmaaraujo committed:
Add Button colors

djalmaaraujo committed:
Adjust button story

luiztiago commented on src/system/Badge/Badge.stories.jsx:
```diff
@@ -8,4 +8,36 @@ export default {
 };
 
-export const Default = () => <Badge>Badge</Badge>;
+const BBadge = props => <Badge { ...props } sx={ { m: 2 } } />;
```
Maybe we can use `const Template` here, such as is used on other files.

luiztiago commented on src/system/Form/Radio.js:
```diff
@@ -10,6 +10,8 @@ import classNames from 'classnames';
 import { screenReaderTextClass } from '../ScreenReaderText/ScreenReaderText';
 import mainTheme from '../theme';
+import { baseControlBorderStyle, inputBaseText } from './Input.styles';
+import { getColor } from '../theme/getColor';
```
Can we remote this line?

luiztiago reviewed (commented):
Great job. The storybook and UI are much better now.
I suggested only two minor updates.

pandah3 reviewed (commented):
What's the best way to test this? I'm not sure where to even start 😅

Should we go through all the components in the [Netlify previews](https://deploy-preview-138--vip-design-system-components.netlify.app/) and compare the colors with the expected colors from the [Valet tokens](https://github.com/Automattic/vip-design-system/blob/trunk/tokens/valet-core.json)? This obviously isn't practical so I'm missing something here lol..

djalmaaraujo commented:
> What's the best way to test this? I'm not sure where to even start 😅
> 
> Should we go through all the components in the [Netlify previews](https://deploy-preview-138--vip-design-system-components.netlify.app/) and compare the colors with the expected colors from the [Valet tokens](https://github.com/Automattic/vip-design-system/blob/trunk/tokens/valet-core.json)? This obviously isn't practical so I'm missing something here lol..

Thanks for the question. You can review all the components in the Netlify preview and make sure they work. We only changed colors; there's no new functionality. 

Some things were removed or deprecated too, but that's minimal, no effect.

I will add a proper way to test in the instructions, but in general, you can also grab the branch into the VIP Dashboard and to some navigation there.

djalmaaraujo committed:
Refactor some things to have better names

djalmaaraujo committed:
Add missing border color

djalmaaraujo committed:
Increase table headers font-size to 14px

djalmaaraujo committed:
Rename colors to be in the singular

djalmaaraujo committed:
Improve Text stories

djalmaaraujo committed:
add missing text helper

djalmaaraujo committed:
Fix code styles

djalmaaraujo committed:
Convert option rows to its own props

djalmaaraujo committed:
Put disabled text as placeholder for now

pandah3 reviewed (changes requested):
Thank you for the updated testing instructions!

Testing with Netlify, a good chunk of the components have accessibility violations in Dark Mode 😢 . These are the ones I found with the violations:

- Accordion
  - Default
- Button
  - Default
  - Link
  - With on click
- Dropdown
  - Default
  - Complex options
  - (With Dialog is good, no violations)
- Option row
  - Default
- Table
  - Default
- Tabs
  - Default
  - (Set active tab is good, no violations)
- Text
  - Default
- Wizard
  - Default
  - (Custom title is good, no violations)
- Autocomplete
  - Default
  - Inline
  - (Others are good)
- Checkbox
  - Default
- Input
  - Default
- Label
  - Required
  - (Default is good, no violations)
- Radio
  - Default
- Select
  - Default
  - With Group
  - Is Inline
  - With Option Label and Value
  - With On Change
- Textarea
  - Default

djalmaaraujo committed:
Set a temporary disabled color for texts

djalmaaraujo committed:
Add a inverse bg to text story

djalmaaraujo commented:
> Thank you for the updated testing instructions!
> 
> Testing with Netlify, a good chunk of the components have accessibility violations in Dark Mode 😢 . These are the ones I found with the violations:
> 
> * Accordion
>   
>   * Default
> * Button
>   
>   * Default
>   * Link
>   * With on click
> * Dropdown
>   
>   * Default
>   * Complex options
>   * (With Dialog is good, no violations)
> * Option row
>   
>   * Default
> * Table
>   
>   * Default
> * Tabs
>   
>   * Default
>   * (Set active tab is good, no violations)
> * Text
>   
>   * Default
> * Wizard
>   
>   * Default
>   * (Custom title is good, no violations)
> * Autocomplete
>   
>   * Default
>   * Inline
>   * (Others are good)
> * Checkbox
>   
>   * Default
> * Input
>   
>   * Default
> * Label
>   
>   * Required
>   * (Default is good, no violations)
> * Radio
>   
>   * Default
> * Select
>   
>   * Default
>   * With Group
>   * Is Inline
>   * With Option Label and Value
>   * With On Change
> * Textarea
>   
>   * Default

Thanks for the reply, but dark mode is not included in the scope of these changes.

djalmaaraujo committed:
Convert table colors to its own props

pandah3 commented:
Noting that this will break Dark Mode even more, so we will be disabling Dark Mode in a separate PR.
https://a8c.slack.com/archives/CKUJZT08Y/p1671744028394699

pandah3 reviewed (changes requested):
Thank you for working on this enormous effort 🎉 
I tested this PR with the [draft Dashboard PR](https://github.com/Automattic/vip-dashboard/pull/2530) using the axe DevTools extension.

Went through all the panels and flows I could in the VIP Dashboard and found some violations of `Ensures the contrast between foreground and background colors meets WCAG 2 AA contrast ratio thresholds`:

1. [Database Backups](http://localhost:3000/apps/4018/production/data/database/backups) - [Axe issue](https://axe.deque.com/issues/fe070724-61fe-4658-9108-6b0aba7e791b
)
```
Element has insufficient color contrast of 3.89 (foreground color: #e74135, background color: #fbfbfb, font size: 10.5pt (14px), font weight: normal). Expected contrast ratio of 4.5:1
```

<img width="809" alt="Screen Shot 2022-12-22 at 5 48 25 PM" src="https://user-images.githubusercontent.com/29221840/209254749-e01ce455-726b-4be7-8bf8-c0a0d32d02d6.png">

2. Media Backups - [Axe issue](https://axe.deque.com/issues/a6d9b72e-c6c4-4d68-94e7-7590e83c6ead)
```
Element has insufficient color contrast of 3.94 (foreground color: #808080, background color: #ffffff, font size: 10.5pt (14px), font weight: normal). Expected contrast ratio of 4.5:1
```
<img width="556" alt="Screen Shot 2022-12-22 at 5 51 56 PM" src="https://user-images.githubusercontent.com/29221840/209255084-a020c8ef-b0ab-4446-939d-c1b8539f2789.png">

3. Deployments - [Axe issue](https://axe.deque.com/issues/fdd904c2-1937-4ab0-820b-5b490904093e)
<img width="549" alt="Screen Shot 2022-12-22 at 5 54 00 PM" src="https://user-images.githubusercontent.com/29221840/209255289-1b092365-1a0a-411d-8bda-7180e21eda16.png">

4. Certs - [Axe issue](https://axe.deque.com/issues/71f2c88a-4132-45e6-a73a-2b528785ede9)
```
Element has insufficient color contrast of 2.58 (foreground color: #fbfbfb, background color: #d29137, font size: 10.0pt (13.3333px), font weight: normal). Expected contrast ratio of 4.5:1
```
<img width="806" alt="Screen Shot 2022-12-22 at 5 55 43 PM" src="https://user-images.githubusercontent.com/29221840/209255681-ba09cce1-13d9-4568-83be-d11e981b1233.png">

5. [Audit log ](http://localhost:3000/apps/5264/production/audit-log?from=2022-12-20T15%3A10%3A37.000Z&eventId=g%3A6617872)- [Axe issue](https://axe.deque.com/issues/e7b864ab-3f43-4222-bf8c-6458308c759d)

<img width="549" alt="Screen Shot 2022-12-22 at 5 58 57 PM" src="https://user-images.githubusercontent.com/29221840/209255826-db2875fb-acf3-44c6-8194-fe3720cb0bab.png">

7. Usage - Axe issues [1](https://axe.deque.com/issues/537378c3-1dc0-4077-8c49-2cd9081a1ab9), [2](https://axe.deque.com/issues/d1d3057a-568c-471d-8ef3-46300d8f9789), [3](https://axe.deque.com/issues/03c6bb7b-7384-4085-97da-dc3b9e91949f)

<img width="379" alt="Screen Shot 2022-12-22 at 6 06 23 PM copy" src="https://user-images.githubusercontent.com/29221840/209256766-fc23c467-2b44-49b2-b1e6-0bdf94d412e5.png">

djalmaaraujo commented:
> Thank you for working on this enormous effort 🎉 I tested this PR with the [draft Dashboard PR](https://github.com/Automattic/vip-dashboard/pull/2530) using the axe DevTools extension.

Thank you for your review @pandah3 

I fixed some issues you mentioned, and some should be fixed in the VIP Dashboard draft PR:

- React Select colors (just a few, because this component is already deprecated), this a PR to replace the one you showed above for the "Type to search…"
- The INFO color (audit log example), is broken, indeed. The color contrast is not the best for the job. I reported to @donaghokeeffe 
- Deployments: Can be fixed in the VIP Dashboard PR (Can be ignored for this PR for now)
- Usage: Can be fixed in the VIP Dashboard PR (Can be ignored for this PR for now)
- Database Backups: Can be fixed in the VIP Dashboard PR (Can be ignored for this PR for now)
- Install Custom Certificate: I found an issue. Fixed here.

djalmaaraujo committed:
Fix input and Deprecated select colors

djalmaaraujo committed:
Fix Code issues with contrast

djalmaaraujo committed:
0.27.12

djalmaaraujo committed:
Adjust Notice stories

andrea-sdl reviewed (approved):
✅ ✅ ✅ ✅ ✅

djalmaaraujo commented:
Merging, and we will fix the blue color of the Notice once @donaghokeeffe send us an updated version of the Valet Theme
